### ISSUE TITLE
Style/RedundantParentheses false positive on one-line rescue within array

### ISSUE THREAD

**Issue by lucthev** [rubocop/rubocop#14386]
When a one-line rescue is used as an array element, v1.79 suggests removing the parentheses, but doing so is a syntax error. Likely as a result of https://github.com/rubocop/rubocop/pull/14355?

```rb
[
  (foo rescue nil),
]
```

--------

## Expected behavior

No `Style/RedundantParentheses` offenses.

## Actual behavior

```
test.rb:2:3: C: [Correctable] Style/RedundantParentheses: Don't use parentheses around a one-line rescue.
  (foo rescue nil),
  ^^^^^^^^^^^^^^^^
```

## Steps to reproduce the problem

```rb
# test.rb
[
  (foo rescue nil),
]
```

```
$ rubocop --cache=false --only Style/RedundantParentheses test.rb
```

## RuboCop version

```
1.79.0 (using Parser 3.3.9.0, Prism 1.4.0, default, rubocop-ast 1.46.0, analyzing as Ruby 3.4, running on ruby 3.4.1) [arm64-darwin23]
```

**Comment by lucthev** [comment]
Tangential, but I ran into this odd caching bug while trying to create the above test-case:

```
$ cat test.rb
[
  (foo rescue nil),
]
$ cp test.rb test.rb.bak
$ rubocop --disable-pending-cops --only Style/RedundantParentheses test.rb -A
Inspecting 1 file
F

Offenses:

test.rb:2:3: C: [Corrected] Style/RedundantParentheses: Don't use parentheses around a one-line rescue.
  (foo rescue nil),
  ^^^^^^^^^^^^^^^^
test.rb:2:6: F: Lint/Syntax: unexpected 'rescue' modifier; expected a , separator for the array elements
(Using Ruby 3.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  foo rescue nil,
     
test.rb:2:7: F: Lint/Syntax: unexpected 'rescue' modifier; expected a ] to close the array
(Using Ruby 3.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  foo rescue nil,
      ^^^^^^
test.rb:2:17: F: Lint/Syntax: unexpected ',', expecting end-of-input
(Using Ruby 3.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  foo rescue nil,
                ^
test.rb:3:1: F: Lint/Syntax: unexpected token ]
(Using Ruby 3.4 parser; configure using TargetRubyVersion parameter, under AllCops)
]
^

1 file inspected, 5 offenses detected, 1 offense corrected
$ mv test.rb.bak test.rb
$ rubocop --disable-pending-cops --only Style/RedundantParentheses test.rb
Inspecting 1 file
F

Offenses:

test.rb:2:3: C: [Correctable] Style/RedundantParentheses: Don't use parentheses around a one-line rescue.
  (foo rescue nil),
  ^^^^^^^^^^^^^^^^
test.rb:2:6: F: Lint/Syntax: unexpected 'rescue' modifier; expected a , separator for the array elements
(Using Ruby 3.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  (foo rescue nil),
     
test.rb:2:7: F: Lint/Syntax: unexpected 'rescue' modifier; expected a ] to close the array
(Using Ruby 3.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  (foo rescue nil),
      ^^^^^^
test.rb:2:17: F: Lint/Syntax: unexpected ',', expecting end-of-input
(Using Ruby 3.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  (foo rescue nil),
                ^
test.rb:2:19: F: Lint/Syntax: unexpected token ]
(Using Ruby 3.4 parser; configure using TargetRubyVersion parameter, under AllCops)
  (foo rescue nil),
                  ^

1 file inspected, 5 offenses detected, 1 offense autocorrectable
```

After failing to auto-correct, the syntax errors get cached as errors on the unmodified file?

**Comment by tagliala** [comment]
Thanks. Was about to post the same issue.

This also happens in Hash elements

```diff
{
-  hello: (world rescue 'nope')
+  hello: world rescue 'nope'
}
```

The (safe) autocorrection results in a syntax error. I'm going to use 

```diff
-  hello: (world rescue 'nope')
+  hello: (world rescue 'nope') # rubocop:disable Style/RedundantParentheses
```

as a workaround for the moment

**Comment by vlad-pisanov** [comment]
Also happens with ternary expressions:

```ruby
cond? ? (foo rescue nil) : 42
```
which gets autocorrected to malformed Ruby:
```ruby
cond? ? foo rescue nil : 42

# syntax error, unexpected `rescue' modifier, expecting ':' (SyntaxError)
```

**Comment by tagliala** [comment]
@koic apologies for the mention, does #14386 also takes into account this use case: https://github.com/rubocop/rubocop/issues/14386#issuecomment-3123895283 ?

**Comment by koic** [comment]
I'll take it and check it later.
